### PR TITLE
Enable Travis deployment for rap-website

### PR DESCRIPTION
**Pull request by willbowditch**
* Added packrat for dependencies, seemed like the easiest way here without having to setup a Description file or long install scripts. Packrat has it's own issues, but probs the easiest way forward here. 
* Travis will build the book every time a commit is made, but will only push the result to github when merged into master.
* The book is now pushed into a separate branch `gh-pages` which can be used to serve the website. Keep the code and results separate. 
* Added a brief how-to in `site/resources-deployment-ci-with-travis.Rmd` this probably needs some edits! 
* Moved the content into a subdir `/site`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+deploy_key
+packrat/lib*/
+
+# Dont put local caches or builds into git
+site/*_cache
+site/*_files
```
Comment (nacnudus) [inline .gitignore]: `site/*_files/`

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,7 @@
+deploy_key
+packrat/lib*/
+
+# Dont put local caches or builds into git
+site/*_cache
```
Comment (nacnudus) [inline .gitignore]: `site/*_cache/`

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+site/*_files
+sites/docs/*
```
Comment (nacnudus) [inline .gitignore]: `site/docs/`

Comment (willbowditch) [reply .gitignore]: Corrected in d3f92a4d179920cacfa86fd409684c52b2447ffd

**Comment by nacnudus** [comment]
Merged, thanks! :tada: 

Except that for some reason GitHub won't actually merge it.  I resolved the conflicts and pushed to master, and it thinks there's still a conflict here even though there clearly isn't.  I'll just have to close it.  Your commits are in master though.

**Pull request closed without merging**